### PR TITLE
Address comments from Audit

### PR DESCRIPTION
**Pull request by birchmd**
Closes https://github.com/aurora-is-near/aurora-engine/issues/76

Addresses most of the minor TODOs from #74 
One still remains, but I'm unsure how to address it. I've [reached out for clarification](https://github.com/aurora-is-near/aurora-engine/pull/74#discussion_r632025663). That last one can be done as a separate PR.

### CODE REVIEW

**Review by sept-en — APPROVED** [review]
LGTM!

Also, as you use constant for the `REGISTER_ID` in `read_input_arr_20()`, please do the same in `read_input()`.

File: src/lib.rs [hunk 1]
```diff
@@ -161,9 +164,7 @@ mod contract {
     /// Call method on the EVM contract.
     #[no_mangle]
     pub extern "C" fn call() {
-        // TODO: Borsh input pattern is so common here. It worth writing sdk::read_input_borsh().
-        let input = sdk::read_input();
-        let args = FunctionCallArgs::try_from_slice(&input).sdk_expect("ERR_ARG_PARSE");
+        let args: FunctionCallArgs = sdk::read_input_borsh().sdk_unwrap();
```
Comment (joshuajbouw) [inline src/lib.rs:167]: Just a general note, I think I was implying the `sdk_unwrap`s to be not really used, unless we are using it for testing or some other purposes. For production, they should always return something clear that we can follow up on.

Comment (birchmd) [reply src/lib.rs:167]: `read_input_borsh` returns a `Result`, so `sdk_unwrap` will use the `AsRef[u8]` implementation on the error type, which returns `ERR_ARG_PARSE` as it did previously. We can change the error message to mention borsh if you think it should be more clear.

**Comment by birchmd** [comment]
> Also, as you use constant for the REGISTER_ID in read_input_arr_20(), please do the same in read_input().

@sept-en done in https://github.com/aurora-is-near/aurora-engine/pull/86/commits/8ae64439b4481839e63f3156fd69b2a6ff75c235

**Review by artob — APPROVED** [review]

**Pull request merged**